stuff/cron/plagiarism_detector.py

```diff
@@ -45,111 +68,210 @@
+
+
 class LocalSubmissionDownloader:
 
     def __init__(self, dir: str) -> None:
-        self._dir = dir
-    
-    def __call__(self, guid: str, destination_path: str) -> None:
-        
-        shutil.copyfile(os.path.join(self._dir, "test1.cpp"), destination_path)
+        self._dir = os.path.join(SUBMISSION_PATH, dir)
+
+    def __call__(self, guid: str, destination_dir: str, language: str) -> None:
+        shutil.copyfile(os.path.join(self._dir, f'{guid[:2]}/{guid[2:]}'),
+                        os.path.join(destination_dir, guid))
+
+
+"""
+    Function to get the range of lines that are plagiarised. 
+    The length of each list should be even. 
+    [1, 4, 7, 9] So, the ranges are 1-4 and 7-9. 
+"""
+
+
 def get_range(code: Sequence[str]) -> Sequence[int]:
 
-    code_range = []
-    for i in range(0,len(code)):
-        if START_RED in code[i]:
-            code_range.append(i)
-        if( START_GREEN in code[i]):
-            code_range.append(i)
-        if END in code[i]:
-            code_range.append(i)
+    code_range: List[int] = []
+    for line_number, line in enumerate(code):
+
+        #If the color is red, then it will be the same for the entire 'code'.
+        #that's why we don't really make a distinction between them.
+
+        if START_RED in line or START_GREEN in line:
+            code_range.append(line_number)
+
+        # We have to check for END separetly so that there is always a end for a start
+        if END in line:
+            code_range.append(line_number)
 
+    # return at one time either the Red lines or green lines range
     return code_range
 
-def filter_and_format_result(dbconn: lib.db.Connection, contest_id: int, submissions: Iterable[Tuple[Any, ...]], result: Sequence[Any]) -> None:
-    d = {}
-    for i in submissions:
-        d[i[5][2:]] = i[1]
-    update = []
-    for r in result:
-        update.append((contest_id, int(100*r[0]), int(100*r[1]),
-                        d[r[2].split('/')[4].split('.')[0]], 
-                        d[r[3].split('/')[4].split('.')[0]],
-                        json.dumps({'file1': get_range(r[4].split('\n')), 
-                                    'file2': get_range(r[4].split('\n'))})
-                        ))
+
+def filter_and_format_result(dbconn: lib.db.Connection, contest_id: int,
+                             submissions: Iterable[Submission],
+                             results: Iterable[CopyDetectorResult]) -> None:
+    """
+        For inserting the result in database we need submission_id, but the result
+        contains guid[2:](the only thing we can have access to from detector). 
+        so we make a dict to map the guid to submission_id from the submissions. 
+    """
+
+    guid_and_submission_id_dict: Dict[str, int] = {}
+
+    for submission in submissions:
+        guid_and_submission_id_dict[submission['guid']
+                                    [2:]] = submission['submission_id']
```

Comment:
```suggestion
                                    ] = submission['submission_id']
```

the files are now using the full guid

Reply:
Done.